docroot/modules/custom/uiowa_core/src/Entity/NodeBundleBase.php

```diff
@@ -71,6 +72,19 @@ public function buildCard(array &$build) {
     if (isset($build['#cache']['keys'])) {
       unset($build['#cache']['keys']);
     }
+
+    // Indicate sticky content if part of view with sticky sort applied.
+    $view = $this->values['view'] ?? NULL;
+    if ($view instanceof ViewExecutable) {
+      $sorts = $view->getDisplay()->getOption('sorts');
+      if (!empty($sorts)) {
+        if (array_key_exists('sticky', $sorts) && $this->isSticky()) {
```

Comment:
```suggestion
    $view = $this?->values['view'] ?? NULL;
    if ($view instanceof ViewExecutable) {
      $sorts = $view?->getDisplay()?->getOption('sorts');
      if (!empty($sorts)) {
        if (array_key_exists('sticky', $sorts) && $this?->isSticky()) {
```

I think that this can be changed for better coding safety?

Otherwise, I think the way you are checking for sticky seems good to me as well, as there is no way that a card will be sticky outside of a view, so checking there is good!

Reply:
Definitely check me on my logic here though, perhaps I am remembering the safety question marks incorrectly

Reply:
At the point this code is called I don't think those additional checks are necessary. The node object ($this) will at least have a `nid` in values and the getDisplay, getOption and isSticky are all safe core functions being called in the right context.

My question is more of should this logic be placed in NodeBundleBase as it is first time we are doing something like this (node card based change that is part of a view that isn't necessarily a listblock) or if it should live elsewhere.

Reply:
Using `?->` and `?? NULL` is redundant. `?->` assigns the value to NULL if it doesn't resolve to a value.

Reply:
I think it should live there. I think it is a fine place.